js/views/message.js

```diff
@@ -70,10 +77,12 @@ views.Message = Backbone.Marionette.ItemView.extend({
 				},
 				type:'POST',
 				success: function () {
-					thisModel.get('flags').set('flagged', !starred);
+					// Retrieve the starred/flag status from the backend and display the results.
+					Mail.UI.messageView.loadMessages();
```

Comment:
This error is most probably related to other code changes in here. I can debug this if you like.
But calling the server is no the solution.